fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractProgressiveLoanScheduleGenerator.java

```diff
@@ -128,6 +145,16 @@ public LoanScheduleModel generate(final MathContext mc, final LoanApplicationTer
 
             if (!isNextRepaymentAvailable) {
                 scheduleParams.getDisburseDetailMap().clear();
+
+                // Fixed Length Validations - If the Number or Repayments requested is greater than current/last
+                // Installment number
+                if (loanApplicationTerms.getFixedLength() != null
```

Comment:
This should be validated in the validator, not during the schedule generation.

Reply:
We have talked about this, that validation will be keep it here

Reply:
Removed from this method and class